homeassistant/components/shelly/config_flow.py

```diff
@@ -177,6 +177,8 @@ async def async_step_confirm_discovery(self, user_input=None):
                 device_info = await validate_input(self.hass, self.host, {})
             except HTTP_CONNECT_ERRORS:
                 errors["base"] = "cannot_connect"
+            except aioshelly.AuthRequired:
+                errors["base"] = "invalid_auth"
```

Comment:
Should we send them to `async_step_credentials` instead?

Reply:
We should, I didn't think about it 😁 I'll change it.